cvs diff -D'20061219 21:07 UTC' -D'20061219 21:10 UTC'
Changes to tests/gkeyfile-test.c not included here
The ones that are are the same in glib-2.12 branch and will be part of glib-2.12.6

http://bugzilla.gnome.org/show_bug.cgi?id=343191
Index: ChangeLog
===================================================================
RCS file: /cvs/gnome/glib/ChangeLog,v
retrieving revision 1.2472
retrieving revision 1.2473
diff -u -B -r1.2472 -r1.2473
--- ChangeLog	19 Dec 2006 15:28:17 -0000	1.2472
+++ ChangeLog	19 Dec 2006 21:08:31 -0000	1.2473
@@ -1,3 +1,11 @@
+2006-12-19  Matthias Clasen  <mclasen@redhat.com>
+
+	* glib/gkeyfile.c (g_key_file_is_key_name): Accept
+	'/', '+' and '.' in key names, since gnome-vfs uses
+	mime types as keys in some cache.
+
+	* tests/keyfile-test.c: Tests for the above.
+
 2006-12-18  Matthias Clasen  <mclasen@redhat.com>
 
 	* configure.in: Fix the broken poll test.  (#387260,
Index: glib/gkeyfile.c
===================================================================
RCS file: /cvs/gnome/glib/glib/gkeyfile.c,v
retrieving revision 1.50
retrieving revision 1.51
diff -u -B -r1.50 -r1.51
--- glib/gkeyfile.c	18 Dec 2006 14:13:33 -0000	1.50
+++ glib/gkeyfile.c	19 Dec 2006 21:08:31 -0000	1.51
@@ -3234,7 +3234,11 @@
     return FALSE;
 
   p = q = (gchar *) name;
-  while (*q && (g_unichar_isalnum (g_utf8_get_char (q)) || *q == '-'))
+  /* We accept a little more than the desktop entry spec says,
+   * since gnome-vfs uses mime-types as keys in its cache.
+   */
+  while (*q && (g_unichar_isalnum (g_utf8_get_char (q)) || 
+                *q == '-' || *q == '_' || *q == '/' || *q == '+' || *q == '.')) 
     q = g_utf8_next_char (q);
   
   if (*q == '[')
